Added postmortem for the Traefik v2 bug

wcarlsen opened the pull request.

samidbb reviewed (commented):
I suggest adding the particular release that we were trying to deploy. I know it's version 2 but which release ?

wcarlsen commented:
@samidbb we didn't change anything. It was a node roll over.

samidbb commented:
okay I think it doesn't matter when the problem has occurred in this case. I suggested mentioning release number because it appeared to be on old issue that was fixed in the next major release, even though I couldn't really find that info by just looking at release notes of 2.5 :) But I still think it could be worth it mentioning which release was running in production.

rifisdfds commented on PM2021-009 - All services returning 404 in Kubernetes.md:
```diff
@@ -0,0 +1,57 @@
+| Time             | Event                   |
+| :--------------- | ----------------------- |
+| 2021-09-01 11:34 | Malformed IngressRoute applied to Kubernetes |
+| 2021-09-06 07:30 | Automated Kuberentes node rollover started |
```
Typo

rifisdfds commented on PM2021-009 - All services returning 404 in Kubernetes.md:
```diff
@@ -0,0 +1,57 @@
+| 2021-09-06 08:05 | Root cause investigation started |
+| 2021-09-06 08:10 | ALB targets unhealthy |
+| 2021-09-06 08:15 | ALB targets healthy after altering the healthcheck path |
+| 2021-09-06 08:20 | All Kubernetes services still returning 404, but now it is Traefik v2 and not the ALB's |
```
I could be mis-remembering but wasn't it Traefik returning 404 the entire time? All of the nodes were marked as unhealthy and when that is the case the ALB still routes traffic (as opposed to if there were 1 or more healthy nodes, it would route to them only). Behaviour is described here - https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-troubleshooting.html#no-healthy-targets

dfds-pwes replied:
Thanks for highlighting the behaviour Richard.  I'd presumed (wrongly) that if there were no healthy targets then it wouldn't route traffic at all, but the link you posted confirms that when all targets go unhealthy it will still route traffic.  So based on that I think your comment is probably correct - the 404's were always coming from Traefik.

rifisdfds reviewed (commented):
I have made a couple of comments

wcarlsen commented:
@rifisdfds and @samidbb I have added a comment on Traefik version < 2.5, corrected the Kubernetes naming typo and corrected the stuff about the ALB's returning 404's instead of Traefik.

rifisdfds reviewed (commented):
Is it intentional that `PM2019-009 - Kubernetes network infrastructure IP starvation.md`  has been edited in this commit/PR?

wcarlsen commented:
Yeah. Formatted the table. What up @rifisdfds ?
